core/corerepo/gc.go

```diff
@@ -66,3 +71,110 @@ func GarbageCollectAsync(n *core.IpfsNode, ctx context.Context) (<-chan *KeyRemo
 	}()
 	return output, nil
 }
+
+func PeriodicGC(ctx context.Context, node *core.IpfsNode) error {
+	r := node.Repo
+	cfg := r.Config()
+	if cfg.Datastore.GCPeriod == "" {
+		r.SetConfigKey("Datastore.GCPeriod", "1h")
+		cfg.Datastore.GCPeriod = "1h"
+	}
+
+	period, err := time.ParseDuration(cfg.Datastore.GCPeriod)
+	if err != nil {
+		return err
+	}
+	if int64(period) == 0 {
+		// if duration is 0, it means GC is disabled.
+		return nil
+	}
+
+	storageMax, storageGC, slackGB, err := getGCParam(node.Repo)
```

Comment:
it's the many params that are a bit unwieldy to pass to functions. maybe ok, just a suggestion.